src/Games/NexusMods.Games.MountAndBlade2Bannerlord/LauncherManager/FileSystemProvider.cs

```diff
@@ -0,0 +1,64 @@
+using Bannerlord.LauncherManager.External;
+using Microsoft.Extensions.Logging;
+
+namespace NexusMods.Games.MountAndBlade2Bannerlord.LauncherManager;
+
+internal sealed class FileSystemProvider : IFileSystemProvider
+{
+    private readonly ILogger _logger;
+
+    public FileSystemProvider(ILogger<FileSystemProvider> logger)
+    {
+        _logger = logger;
+    }
+
+    public byte[]? ReadFileContent(string filePath, int offset, int length)
```

Comment:
I don't mind personally either way since the paths are sourced from the `LauncherManager` module, but I know the other devs would complain if they reviewed this, so I'll bring it up.

Please wrap the string in `NexusMods.Paths` with `AbsolutePath`.
Namely, inject an `IFileSystem` via the constructor, and create a path with `_fs.FromUnsanitizedFullPath(filePath)`. Then perform the read from that.

Reply:
My only concern is that we'll have to switch to Streams. It's a shame that we won't be able to use `RandomAccess.Read(handle, data, offset);`

Reply:
I'm fine with this code personally, and we could leave it this way, since the paths are provided by the library in any case; so the whole magic of abstraction is lost.
I just more-so worry about what the others would think.